server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -285,13 +287,20 @@ public void createIndex(final CreateIndexClusterStateUpdateRequest request, fina
 
     private void onlyCreateIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<AcknowledgedResponse> listener) {
         normalizeRequestSetting(request);
+
+        var future = new ListenableFuture<Void>();
+
         submitUnbatchedTask(
             "create-index [" + request.index() + "], cause [" + request.cause() + "]",
-            new AckedClusterStateUpdateTask(Priority.URGENT, request, listener) {
+            new AckedClusterStateUpdateTask(
+                Priority.URGENT,
+                request,
+                future.delegateFailure((delegate, response) -> future.addListener(listener.map(ignored -> response)))
```

Comment:
Should this be `listener.delegateFailure`? We want to complete `listener` on a failure, we don't care about `future`.